src/Databases/PostgreSQL/FetchFromPostgreSQL.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#if !defined(ARCADIA_BUILD)
+#include "config_core.h"
+#endif
+
+#if USE_LIBPQXX
+#include <Storages/StoragePostgreSQL.h>
+
+namespace DB
+{
+
+std::shared_ptr<NamesAndTypesList> fetchTableStructure(ConnectionPtr connection, const String & postgres_table_name, bool use_nulls);
```

Comment:
ConnectionPtr also?

Reply:
Where? In function arguments it is already ConnectionPtr and in function result it is a different type.